### PR TITLE
Remove unused git submodules

### PR DESCRIPTION
**Pull request by snnn**

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -584,10 +584,41 @@ if (onnxruntime_USE_CANN)
     list(APPEND ONNXRUNTIME_PROVIDER_NAMES cann)
 endif()
 
+if (onnxruntime_ENABLE_EAGER_MODE)
+  if (NOT onnxruntime_ENABLE_TRAINING OR NOT onnxruntime_ENABLE_PYTHON)
+    message(
+        FATAL_ERROR
+        "Option onnxruntime_ENABLE_EAGER_MODE can only be used when onnxruntime_ENABLE_TRAINING and onnxruntime_ENABLE_PYTHON are enabled")
+  endif()
+  add_compile_definitions(ENABLE_EAGER_MODE)
```
Comment (snnn) [inline cmake/CMakeLists.txt:593]: TODO: remove this line.

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -584,10 +584,41 @@ if (onnxruntime_USE_CANN)
...
+
+if (onnxruntime_ENABLE_LAZY_TENSOR)
+  # To support LazyTensor, ORT needs to call Python function from C/C++.
+  # so onnxruntime_ENABLE_PYTHON is required.
+  if (NOT onnxruntime_ENABLE_TRAINING OR NOT onnxruntime_ENABLE_PYTHON)
+    message(
+        FATAL_ERROR
+        "Option onnxruntime_ENABLE_LAZY_TENSOR can only be set when onnxruntime_ENABLE_TRAINING and onnxruntime_ENABLE_PYTHON are enabled")
+  endif()
+  add_compile_definitions(ENABLE_LAZY_TENSOR)
```
Comment (snnn) [inline cmake/CMakeLists.txt:605]: TODO: remove this line.

**Comment by jywu-msft** [comment]
for deps.txt, can we adopt a convention that the zip urls should refer to a tag or branches?
e.g. onnx-tensorrt url should be changed to https://github.com/onnx/onnx-tensorrt/archive/refs/tags/release/8.4-GA.zip ?

**Comment by jywu-msft** [comment]
> > for deps.txt, can we adopt a convention that the zip urls should refer to a tag or branches rather than commitid? e.g. onnx-tensorrt url should be changed to https://github.com/onnx/onnx-tensorrt/archive/refs/tags/release/8.4-GA.zip ?
> 
> Yes, we can. But the one we are using doesn't match the release tag. It doesn't match to any tag.

according to .gitmodules we were using branch 8.4-GA which is the same as release/8.4-GA
https://github.com/microsoft/onnxruntime/blob/65201e47bf58ce5bbc8389a83082f69ee1ea50b3/.gitmodules#L44

**Pull request merged**